server/logger.js

```diff
@@ -0,0 +1,100 @@
+import winston from 'winston';
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+import path from 'path';
+
+const { combine, timestamp, printf, colorize } = winston.format;
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const ROOT = __dirname;
+
+// Apply colors to log levels
+const colorizer = colorize({ all: true });
+
+const myFormat_console = printf(({ level, message, timestamp }) => {
+  return `${timestamp} ${level}: ${message}`;
+});
+
+const logger = new winston.createLogger({
+  level: 'info',
+  defaultMeta: { service: 'user-service' },
+  transports: [
+    new winston.transports.Console({
+      format: combine(colorizer, timestamp(), myFormat_console),
+    }),
+    new winston.transports.File({
+      filename: 'server.log',
+      format: combine(timestamp(), myFormat_console),
+    }),
+  ],
+  exitOnError: false,
+});
+
+fs.writeFile('server.log', '', () => {});
```

Comment:
The `fs.writeFile` call should handle errors properly to ensure that any issues with file writing are logged or managed. 
```diff
- fs.writeFile('server.log', '', () => {});
+ fs.writeFile('server.log', '', (err) => { if (err) logger.error('Error writing to log file', err); });
```
*This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*